staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -269,37 +300,46 @@ func (cfgCtlr *configController) processNextWorkItem() bool {
 
 	func(obj interface{}) {
 		defer cfgCtlr.configQueue.Done(obj)
-		if !cfgCtlr.syncOne() {
+		_, _, needRetry, wait := cfgCtlr.SyncOne()
+		switch {
+		case needRetry:
 			cfgCtlr.configQueue.AddRateLimited(obj)
-		} else {
+		case wait > 0:
+			cfgCtlr.configQueue.AddAfter(obj, wait)
+		default:
 			cfgCtlr.configQueue.Forget(obj)
 		}
 	}(obj)
 
 	return true
 }
 
-// syncOne attempts to sync all the API Priority and Fairness config
-// objects.  It either succeeds and returns `true` or logs an error
-// and returns `false`.
-func (cfgCtlr *configController) syncOne() bool {
+// SyncOne does one full synchronization.  It reads all the API
+// objects that configure API Priority and Fairness and updates the
+// local configController accordingly.  The return values are:
+// - triedWrites bool indicating whether status updates were attempted;
+// - didWrites bool indicating whether status udates were successful;
+// - needRetry bool indicating whether to requeue with exponential backoff;
+// - wait time.Duration indicating to requeue after a specific delay.
+func (cfgCtlr *configController) SyncOne() (bool, bool, bool, time.Duration) {
```

Comment:
Please at least name the return parameters. But it's even better to make a "report" struct object and return that.

Reply:
Made a report structure.